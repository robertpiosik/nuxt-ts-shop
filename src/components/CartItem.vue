<template lang="pug">
	div(:class="$style.container")
		div(:class="$style.thumbnail")
			img(:src="thumbnail")
		div(:class="$style.title") {{ title }}
		div(:class="$style.quantity") {{ quantity }}
		div(:class="$style.priceAndTotal")
			div(:class="$style.price")
				span(:class="$style.priceValue") {{ priceFormatted.value }}
				span(:class="$style.pricePenny") {{ priceFormatted.penny }}
			div(:class="$style.total")
				span(:class="$style.priceValue") {{ totalFormatted.value }}
				span(:class="$style.pricePenny") {{ totalFormatted.penny }}
</template>

<script lang="ts">
import { Vue, Component, Prop } from 'vue-property-decorator'

import { priceFormatter } from './../utils/transformations'

@Component
export default class CartItem extends Vue {
  @Prop({ type: Number, required: true }) readonly id!: number
  @Prop({ type: Number, required: true }) readonly quantity!: number
  @Prop({ type: String, required: true }) readonly title!: string
  @Prop({ type: Number, required: true }) readonly price!: number
  @Prop({ type: String, required: true }) readonly thumbnail!: string

  get totalFormatted() {
    return priceFormatter(this.price * this.quantity)
  }
  get priceFormatted() {
    return priceFormatter(this.price)
  }
}
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
  @include atSmall {
    flex-direction: row;
  }
}
.thumbnail {
  position: relative;
  @include toExtraSmall {
    width: 7rem;
    height: 7rem;
  }
  width: 8rem;
  height: 8rem;
  overflow: hidden;
  @include atSmall {
    width: 10rem;
    height: 10rem;
  }
  & img {
    width: 100%;
    height: 100%;
    position: absolute;
    object-fit: cover;
  }
}
.title {
  display: flex;
  align-items: center;
  margin-left: 1.5rem;
  width: 20rem;
}
.quantity {
  display: flex;
  width: 10rem;
}
.priceAndTotal {
}
</style>
