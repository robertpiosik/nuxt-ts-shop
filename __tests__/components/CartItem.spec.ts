import { mount } from '@vue/test-utils'

import CartItem from './../src/components/CartItem.vue'

const wrapper = mount(CartItem)

describe('CartItem', () => {
  describe('GENERAL', () => {
    it('should be a Vue instance', () =>
      expect(wrapper.isVueInstance).toBeTruthy)
  })
})
