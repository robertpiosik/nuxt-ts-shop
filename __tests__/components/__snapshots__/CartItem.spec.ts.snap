// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CartItem GENERAL should match a snapshot 1`] = `
<div class="container">
  <div class="thumbnail"><img src="someUrl"></div>
  <div class="title">Water</div>
  <div class="quantity">
    <div>2</div>
    <div>-</div>
    <div>+</div>
  </div>
  <div class="priceAndTotal">
    <div>1,2 zł</div>
    <div>2,4 zł</div>
  </div>
</div>
`;
